lib/Whim/Core.pm

```diff
@@ -202,15 +201,10 @@ sub process_webmentions( $self ) {
         # Grab the author image
         my $photo_hash;
         if ( $wm->author && $wm->author->photo ) {
-            my $url = $wm->author->photo->as_string;
-            $self->ua->get_p($url)->then(
-                sub( $tx ) {    # Promise accepted
-                    $photo_hash = $self->_process_author_photo_tx($tx);
-                },
-                sub( @args ) {    # Promise rejected
-                    warn "Couldn't get author photo from $url: @args\n";
-                }
-            )->wait;
+            my $url      = $wm->author->photo->abs( $wm->source )->as_string;
+            my $response = $self->ua->get($url);
+            $photo_hash = $self->_process_author_photo_tx($response);
+            $wm->author_photo_hash($photo_hash);
```

Comment:
I wonder if you should be checking for definededness here?

Reply:
I don't think so... if $photo_hash is undef then it's OK to assign undef to author_photo_hash (and the new attribute's `isa` explicitly allows this).